f8a_report/manifest_helper.py

```diff
@@ -0,0 +1,110 @@
+"""Dynamic Generation of Manifest files."""
+
+import random
+import json
+from s3_helper import S3Helper
+import logging
+import os
+import re
+
+logger = logging.getLogger(__file__)
+logging.basicConfig(level=logging.INFO)
+
+
+class GetReport:
+    """This creates a manifest file for all ecosystem and save to s3."""
+
+    def __init__(self):
+        """Init method for the Report helper class."""
+        self.s3 = S3Helper()
+        self.curr_dir = os.path.join(
+            os.path.abspath(os.curdir), "f8a_report", "manifests")
+        if not os.path.exists(self.curr_dir):
+            os.makedirs(self.curr_dir)
+
+    def generate_manifest_for_pypi(self, stack_report):
+        """Generate manifest file for pypi."""
+        logger.info('Generating Manifest for Pypi executed')
+        file_name = "pylist.json"
+        file_path = os.path.join(self.curr_dir, file_name)
+        with open(file_path, 'w') as manifest:
+            json.dump(stack_report, manifest)
+        return self.save_manifest_to_s3(file_path=file_path, file_name=file_name)
+
+    def generate_manifest_for_npm(self, stack_report):
+        """Generate manifest file for npm."""
+        logger.info('Generating manifest for NPM executed')
+        file_name = "npmlist.json"
+        file_path = os.path.join(self.curr_dir, file_name)
+        with open(file_path, 'w') as manifest:
+            json.dump(stack_report, manifest)
+        return self.save_manifest_to_s3(file_path=file_path, file_name=file_name)
+
+    def generate_manifest_for_maven(self, stack_report):
+        """Generate manifest file for maven."""
+        logger.info('Generate Manifest for Maven executed')
+        file_name = "dependencies.txt"
+        file_path = os.path.join(self.curr_dir, file_name)
+        with open(file_path, 'w') as manifest:
+            manifest.write(stack_report)
+        return self.save_manifest_to_s3(file_path=file_path, file_name=file_name)
+
+    def save_manifest_to_s3(self, file_path, file_name):
+        """Save Generated manifest file to S3."""
+        manifest_file_key = f'dynamic_manifests/{file_name}'
+        self.s3.store_file_object(file_path=file_path,
+                                  bucket_name=self.s3.report_bucket_name,
+                                  file_name=manifest_file_key)
+
+
+class FilterStacks:
+    """This filters a Manifest file from collated stack report."""
+
+    def filter_stacks_on_ecosystem(self, stack_report, stack_size):
+        """Filter Stack Report on ecosystem."""
+        logger.info('Filtering Stacks on ecosystem executed')
+        npm_stack_data = []
+        pypi_stack_data = []
+        maven_stack_data = []
+        for stack in stack_report:
+            data = stack[0]['manifest'][0]
+
+            if data['ecosystem'] == 'npm':
+                npm_stack_data.append(data)
```

Comment:
You are appending all the data to an array list and then randomly selecting 2 later. Rather just add those random samples directly.

Reply:
@deepak1725 - as commented earlier, this can be optimized further by selecting `filter_stacks_on_size` on `data` directly rather than creating `npm_stack_data`

Reply:
Data is loaded from `dict` format. Here we are making 3 lists of stacks for each ecosystem to select randomly on top of it